### DISCUSSION TITLE
Questions regarding image generation for jetson orin nano devkit using nvme

### DISCUSSION THREAD

**Discussion by reymor** [OE4T/meta-tegra#1502] [Q&A]
Hi everyone,

I have a question regarding the `nvme` flashing in jetson orin nano devkit

We are working with `kirkstone` branch --> `b7a0792b996b47305c7ec1713621b0e07d314700`

We build a tegraflash with the machine `jetson-orin-nano-devkit-nvme`. Is this enough for getting a compatible image with jetson-orin-nano-devkit-nvme or there something more we shall set/specify?

I do this a try just `doflash.sh` over tegraflash directory

Doing so, just flash the spi leaving the nvme without any change --> Like this https://github.com/OE4T/meta-tegra/discussions/1334

I realized about the initrd and tried also that, but I got something similar to this: https://github.com/OE4T/meta-tegra/discussions/1350

In which it shows:

```
== Step 4: Writing partitions on external storage device at ==
Waiting for USB storage device nvme0n1 from ..[got flashpkg when expecting nvme0n1]
ERR: could not find nvme0n1
```

I saw the ideas shared by madisongh, but in my case it does not seem the same problem because I did the inspection of the exposed device and it is always a `flashpkg` which is exactly what the scripts seems and it does not seem to be related with the races, I just test what he said about the delay loop without any hope and as I was expected it does not work. 

So, at this point I checked the `flash.xml.in` which does not include information related with nvme, I manually added the information and I was able to flash the nvme with `doflash.sh` script. I am not really sure if there is something that I need to add in local.conf, this is not something related with https://github.com/OE4T/meta-tegra/discussions/1350 because this variable seems that was added in current commits.

I have question regarding the ways of doing so (flashing the nvme). The idea is using the initrd, or is there any plan to support both? which problems did you face using the `doflash.sh` way?

Thanks in advance,

**Comment by madisongh** [comment]
The `[got flashpkg when expecting nvme0n1]` message means there was probably a failure on the device side when it was trying to export the NVME drive.  (The script should handle this case better.) Can you monitor the serial console?  That should tell us more about what's going on there.
